src/main/java/io/fabric8/maven/docker/SaveMojo.java

```diff
@@ -101,6 +104,10 @@ private String getImageName() throws MojoExecutionException {
 		// specify image by name or alias
 		if (saveName == null && saveAlias == null) {
 			List<ImageConfiguration> buildImages = getImagesWithBuildConfig(images);
+			if (buildImages.isEmpty()) {
```

Comment:
Can we move that check up to the start of `executeInternal()` ? E.g. we can obtain the configuration there and add it as an argument to this method.

Its much clearer to skip execution early if needed.